src/ui/layout/tree-item.vue

```diff
@@ -137,8 +132,15 @@ export default {
         }
 
         this.openmct.router.on('change:path', this.highlightIfNavigated);
+    },
+    updated() {
         if (this.emitHeight) {
-            this.$emit('emittedHeight', this.$refs.me);
+            this.$nextTick(() => {
```

Comment:
A `nextTick` should not be needed as the first line of an `updated` call. By definition, `updated` will only be called after a tick has already taken place, and since the state has not changed yet (in fact `updated` should never mutate state) `nextTick` should not be needed.

I'm a little concerned about the number of nextTicks and setTimeouts in this code, and I wonder whether we could refactor this in future to use an event-driven model. (Perhaps [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) might come in handy?). Don't address it now, I'll file a followup.

Reply:
Removed that nextTick as it was unnecessary, unfortunately the timeout is. Would love a more reliable way to get fully rendered DOM element information.

Reply:
I take back my earlier statement, there are some cases where calling `$nextTick()` immediately inside updated makes sense, as it ensures that all children have also rendered.